mopidy/core/tracklist.py

```diff
@@ -381,7 +397,21 @@ def add(self, tracks=None, at_position=None, uris=None):
 
         tracks is None or validation.check_instances(tracks, Track)
         uris is None or validation.check_uris(uris)
-        validation.check_integer(at_position or 0)
+
+        if isinstance(at_position, (str, AddAtPosition)):
+            at_position = validation.check_enum(
+                at_position,
+                AddAtPosition,
+                msg="Expected an integer, or a member of {enum_name}: {enum_members}",
+            )
+        elif at_position is not None and not isinstance(at_position, int):
+            err_msg = "Expected an integer, or a member of {enum_name}: {enum_members}"
```

Comment:
If you move this up above the if statement you can reuse it when calling `validation.check_enum()`.